sno/working_copy/gpkg.py

```diff
@@ -616,20 +649,21 @@ def write_full(self, target_tree_or_commit, *datasets, safe=True):
                 self.write_meta(dataset)
 
                 # Create the table
-                cols, pk_field = self._get_columns(dataset)
-                col_names = cols.keys()
-                col_specs = cols.values()
-                dbcur.execute(
-                    f"""
-                    CREATE TABLE {gpkg.ident(table)}
-                    ({', '.join(col_specs)});
-                """
-                )
+                table_spec = gpkg_adapter.v2_schema_to_sqlite_spec(dataset)
+
+                # GPKG requires an integer primary key for spatial tables, so we add it in if needed:
+                if dataset.has_geometry and "PRIMARY KEY" not in table_spec:
+                    table_spec = (
+                        '"integer-pk" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,'
```

Comment:
maybe `".sno-generated-pk"` would be better so it's clearer where it came from

Reply:
Sort-of-done - changed to `sno-auto-pk`. What do you think?

Nuance: an "auto-pk" sounds to me more like something temporary and disposable. A "generated-pk" sounds to me more like something we expended some effort creating for the user, and maybe like something we could regenerate consistently (but we can't, it's just some random number).